### ISSUE TITLE
Architecture of valuenet

### ISSUE THREAD

**Issue by wlike** [Rochester-NRT/RocAlphaGo#110]
Do you check the correctness of the value network?
According to the code, hidden layers 2 to 12 are identical to that of policy network, rather than “2 to 11”, stated in the paper.

**Comment by wrongu** [comment]
The paper is confusingly worded. They say "Hidden layers 2 to 11 are identical to the policy network, hidden layer 12 is an additional convolution layer..."

Thanks for looking at it carefully - let us know if you find anything else that is suspect :)

**Comment by wlike** [comment]
Do you train the value network using KGS dataset? Does the mse approach that described in the paper?
I've been training the value network. The mse of training set can quickly down to 0.1, but the mse of validation set remains 0.5~0.6.

**Comment by wrongu** [comment]
Unfortunately we haven't tried training the value network yet.. We only recently got into the 45% range with supervised training accuracy.

The value network isn't trained directly on a dataset. To clarify, we're doing supervised learning using the GoGoD dataset, then the value network is trained based on games of self-play when the trained policy network plays against itself.

It could be that if you're training the value net directly on the KGS dataset (I assume you're training it to map from a given position to the end result of the game) then you may be experiencing this problem that Silver et al refer to in the paper:

> The naive approach of predicting game outcomes from data consisting of complete games leads to overfitting. The problem is that successive positions are strongly correlated, differing by just one stone, but the regression target is shared for the entire game. When trained on the KGS data set in this way, the value network memorized the game outcomes rather than generalizing to new positions, achieving a minimum MSE of 0.37 on the test set, compared to 0.19 on the training set...

Then they go on to explain how they generate a new dataset for training that helps "mitigate" that problem.

**Comment by wlike** [comment]
There is something wrong with the architecture of the value network, according to the paper "hidden layer 14 is a fully connected linear layer with 256 rectifier units"

**Comment by wrongu** [comment]
@wlike with the last commit, is the value network looking like what you interpreted from the paper?

**Comment by wlike** [comment]
Yes, at least for now.

**Comment by wlike** [comment]
@wrongu yes, at least for now.
